python/revng/pypeline/daemon/daemon.py

```diff
@@ -64,30 +61,31 @@ def __init__(
         self.pipeline = pipeline
         self.debug = debug
         self.cache_dir = cache_dir
-
-        self.project_data: Locked[tuple[dict[ProjectID, int], StorageProviderFactory]] = Locked(
-            (defaultdict(lambda: 0), storage_provider_factory_factory(storage_provider_url))
-        )
+        self.storage_provider_factory = storage_provider_factory_factory(storage_provider_url)
 
     async def get_epoch(self, request) -> Response:
-        async with self.project_data() as (epochs, _):
-            return Response(code=200, body={"epoch": epochs[request["project_id"]]})
+        project_id = request["project_id"]
+        token = request.get("token")
+        storage_provider_context = self.storage_provider_factory.get(
+            project_id=project_id,
+            token=token,
+            cache_dir=self.cache_dir,
+        )
+        async with storage_provider_context as storage_provider:
+            return Response(code=200, body={"epoch": storage_provider.get_model()[1]})
```

Comment:
`get_model()` can be quite expensive (e.g. a non-trivial model is in the 100s of kilobyte range), transfering all of it to then just return an int is quite wasteful. I'd have a separate `get_epoch` in the storage provider that just returns the int